### PR TITLE
Docker test framework updates

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: docker-test-framework/4-3/Dockerfile.consumer [hunk 1]
```diff
@@ -1,30 +1,55 @@
...
     apt-get update && \
     apt-get install -y \
         irods-runtime=${irods_package_version} \
         irods-icommands=${irods_package_version} \
-        irods-server=${irods_package_version}
+        irods-server=${irods_package_version} \
+    && \
+    rm -rf /tmp/*
 
 EXPOSE 1247 1248
```
Comment (trel) [inline docker-test-framework/4-3/Dockerfile.consumer:49]: do we still need this?

File: docker-test-framework/4-3/Dockerfile.provider [hunk 2]
```diff
@@ -1,39 +1,68 @@
...
     apt-get install -y \
         irods-runtime=${irods_package_version} \
         irods-icommands=${irods_package_version} \
         irods-server=${irods_package_version} \
-        irods-database-plugin-postgres=${irods_package_version}
+        irods-database-plugin-postgres=${irods_package_version} \
+    && \
+    rm -rf /tmp/*
 
 EXPOSE 1247 1248
```
Comment (trel) [inline docker-test-framework/4-3/Dockerfile.provider:60]: same

File: docker-test-framework/4-3/Dockerfile.consumer [hunk 3]
```diff
@@ -1,30 +1,55 @@
...
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    apt-get update && \
+    apt-get install -y \
+        sudo \
+        wget \
+        less \
+        lsb-release \
+        apt-transport-https \
+        netcat \
+        gnupg2 \
```
Comment (alanking) [inline docker-test-framework/4-3/Dockerfile.consumer]: I'm guessing we don't need `less`, but I could be wrong.

File: docker-test-framework/4-3/Dockerfile.provider [hunk 4]
```diff
@@ -1,39 +1,68 @@
...
+        lsb-release \
+        apt-transport-https \
+        postgresql \
+        vim \
+        python3-pip \
+        libfuse2 \
+        unixodbc \
+        rsyslog \
+        less \
+        gnupg2 \
```
Comment (alanking) [inline docker-test-framework/4-3/Dockerfile.provider]: We probably don't need `vim` or `less` (again, could be wrong).

File: docker-test-framework/4-3/Dockerfile.consumer (hunk 1 above)
Comment (korydraughn) [reply docker-test-framework/4-3/Dockerfile.consumer:49]: We do that in nearly all the dockerfiles for the development environment.

I believe that helps keep the images as small as possible.

File: docker-test-framework/4-3/Dockerfile.provider (hunk 2 above)
Comment (korydraughn) [reply docker-test-framework/4-3/Dockerfile.provider:60]: See https://github.com/DICE-UNC/jargon/pull/475#discussion_r1732902980

File: docker-test-framework/4-3/Dockerfile.consumer (hunk 3 above)
Comment (korydraughn) [reply docker-test-framework/4-3/Dockerfile.consumer]: I doubt we need it, but I'm not sure. Those packages have always been there.

Will remove and see what happens.

File: docker-test-framework/4-3/Dockerfile.provider (hunk 4 above)
Comment (korydraughn) [reply docker-test-framework/4-3/Dockerfile.provider]: See https://github.com/DICE-UNC/jargon/pull/475#discussion_r1732906618

File: docker-test-framework/4-3/Dockerfile.consumer (hunk 3 above)
Comment (korydraughn) [reply docker-test-framework/4-3/Dockerfile.consumer]: Removed `less`.

File: docker-test-framework/4-3/Dockerfile.provider (hunk 4 above)
Comment (korydraughn) [reply docker-test-framework/4-3/Dockerfile.provider]: Removed `less` and `vim`.

**Comment by korydraughn** [comment]
Running test suite to see if anything breaks.

**Review by alanking — APPROVED** [review]
Squash and # and merge once tests are confirmed passing, if you're happy with it.

**Comment by korydraughn** [comment]
All tests passed against 4.3.3 (this PR sets the default to the latest version of iRODS).

Once this PR is merged, Jargon will be ready for a release.

**Comment by korydraughn** [comment]
Pounded. Merging.

**Pull request merged**